utils/zerovec/src/zerovec/mod.rs

```diff
@@ -23,48 +23,63 @@ use core::ops::Deref;
-/// items from `ZeroVec<T>`, we fetch the `T::ULE`, convert it on the fly to `T`, and return `T` by
-/// value.
-///
-/// Benchmarks can be found in the project repository. We found that for common operations on small
-/// and large vectors, `ZeroVec<T>` performs from slightly faster to 15% slower than `Vec<T>`.
-/// However, the main performance improvement on `ZeroVec<T>` is when deserializing from a byte
-/// array; `ZeroVec<T>` deserializes 80% faster than `Vec<T>` in Serde Bincode, and it does not
-/// require any heap allocations.
-///
-/// # Safety
-///
-/// `ZeroVec<T>` contains no unsafe code. However, the conversion from `&[u8]` to `&[T::ULE]` may
-/// be unsafe. For more information, see the [`ule`] module.
+/// `ZeroVec<T>` behaves much like [`Cow`](alloc::borrow::Cow), where it can be constructed from
+/// owned data (and then mutated!) but can also borrow from some buffer.
 ///
 /// # Example
 ///
 /// ```
 /// use zerovec::ZeroVec;
 ///
 /// // The little-endian bytes correspond to the numbers on the following line.
-/// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
 /// let nums: &[u16] = &[211, 281, 421, 461];
 ///
-/// let zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).unwrap();
+/// #[derive(serde::Serialize, serde::Deserialize)]
+/// struct Data<'a> {
+///     #[serde(borrow)]
+///     nums: ZeroVec<'a, u16>,
+/// }
+///
+/// // The owned version will allocate
+/// let data = Data { nums: ZeroVec::alloc_from_slice(nums) };
+/// let bincode_bytes = bincode::serialize(&data)
+///     .expect("Serialization should be successful");
+///
+/// // Will deserialize without allocations
+/// let deserialized: Data = bincode::deserialize(&bincode_bytes)
+///     .expect("Deserialization should be successful");
 ///
-/// assert!(matches!(zerovec, ZeroVec::Borrowed(_)));
-/// assert_eq!(zerovec.get(2), Some(421));
-/// assert_eq!(zerovec, nums);
+/// // This deserializes without allocation!
+/// assert!(matches!(deserialized.nums, ZeroVec::Borrowed(_)));
+/// assert_eq!(deserialized.nums.get(2), Some(421));
+/// assert_eq!(deserialized.nums, nums);
 /// ```
 ///
 /// [`ule`]: crate::ule
-#[non_exhaustive]
+///
+/// # How it Works
+///
+/// `ZeroVec<T>` represents a slice of `T` as a slice of `T::ULE`. The difference between `T` and
+/// `T::ULE` is that `T::ULE` must be encoded in little-endian with 1-byte alignment. When accessing
+/// items from `ZeroVec<T>`, we fetch the `T::ULE`, convert it on the fly to `T`, and return `T` by
+/// value.
+///
+/// Benchmarks can be found in the project repository. We found that for common operations on small
+/// and large vectors, `ZeroVec<T>` performs from slightly faster to 15% slower than `Vec<T>`.
+/// However, the main performance improvement on `ZeroVec<T>` is when deserializing from a byte
+/// array; `ZeroVec<T>` deserializes 80% faster than `Vec<T>` in Serde Bincode, and it does not
+/// require any heap allocations.
+///
```

Comment:
Nit: Just point to the main doc with the table of benchmarks rather than editorializing them.